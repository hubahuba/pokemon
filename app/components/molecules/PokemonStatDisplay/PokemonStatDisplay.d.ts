export interface PokemonStatProps {
  name: string;
  value: number;
}
