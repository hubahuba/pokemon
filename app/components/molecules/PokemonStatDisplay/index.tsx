import * as React from 'react';
import {View, Text} from 'react-native';
import * as Progress from 'react-native-progress';
import {PokemonStatProps} from './PokemonStatDisplay';

export default function PokemonStatDisplay({name, value}: PokemonStatProps) {
  return (
    <View className="flex flex-row space-x-2">
      <View className="basis-[130px] items-end">
        <Text className="text-[16px] font-[Raleway-Medium]">{name}</Text>
      </View>
      <View className="grow justify-center">
        <Progress.Bar width={190} borderRadius={2} progress={value} />
      </View>
    </View>
  );
}
