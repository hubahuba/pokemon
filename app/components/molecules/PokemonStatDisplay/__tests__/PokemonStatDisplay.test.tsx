import React from 'react';
import renderer from 'react-test-renderer';
import PokemonStatDisplay from '../index';

it('renders Card correctly', () => {
  const tree = renderer
    .create(<PokemonStatDisplay name="test stat" value={2} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
