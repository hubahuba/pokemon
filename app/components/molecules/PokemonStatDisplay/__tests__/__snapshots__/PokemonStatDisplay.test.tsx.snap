// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card correctly 1`] = `
<View
  style={
    [
      {
        "display": "flex",
      },
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": 130,
        },
        {
          "alignItems": "flex-end",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#475569",
          },
          {
            "fontSize": 16,
          },
          {
            "fontFamily": "Raleway-Medium",
          },
        ]
      }
    >
      test stat
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexGrow": 1,
        },
        {
          "justifyContent": "center",
        },
        [
          {
            "marginLeft": 8,
          },
        ],
      ]
    }
  >
    <View
      animated={true}
      animationConfig={
        {
          "bounciness": 0,
        }
      }
      animationType="spring"
      indeterminate={false}
      indeterminateAnimationDuration={1000}
      onLayout={[Function]}
      progress={2}
      style={
        [
          {
            "backgroundColor": undefined,
            "borderColor": "rgba(0, 122, 255, 1)",
            "borderRadius": 2,
            "borderWidth": 1,
            "overflow": "hidden",
            "width": 190,
          },
          undefined,
        ]
      }
      useNativeDriver={false}
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(0, 122, 255, 1)",
            "height": 6,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateX": 0,
              },
              {
                "scaleX": 1,
              },
            ],
          }
        }
      />
    </View>
  </View>
</View>
`;
